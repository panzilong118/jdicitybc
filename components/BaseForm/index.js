/**
 * form表单 base版
 * props:
 * @param {object} layout { labelCol: { span }, wrapperCol: { span } }
 * @param {array} items
 *    items=[{
 *      col,  // column
 *      label,
 *      key,
 *      required,
 *      component, // default is Input
 *      options = {},
 *      rules
 *    }]
 *
 * usage:
 *  import { Form } from 'src/components/common'
 *
 *  add = (e) =>{
 *    const { onOk } = this.props;
 *    Form.submit(e, this.formRef, (values) => {
 *      console.log('valide', values)
 *      // post to server
 *      onOk()
 *    })
 *  }
 *
 *  // render
 *  <Form
 *    items={[{
 *      label: '目录名',
 *      required: true,
 *      key: 'name'
 *    }, {
 *      label: '是否挂表',
 *      key: 'isLast',
 *      component: (
 *        <Switch
 *          checkedChildren={<Icon type="check" />}
 *          unCheckedChildren={<Icon type="close" />}
 *        />
 *      )
 *    }]}
 *  />
*/
import React from 'react';
import { Form, Input, Col } from 'antd';

const FormItem = Form.Item;
// 默认的layout
export const defaultLabelColSpan = 6;
const defaultFormItemLayout = {
  labelCol: { span: defaultLabelColSpan },
  wrapperCol: { span: 16 },
};

class BaseForm extends React.Component {
  static submit({ validateFieldsAndScroll, resetFields }, onValid) {
    validateFieldsAndScroll(async (err, values) => {
      if (err) {
        return;
      }
      await onValid(values);
      resetFields();
    });
  }

  // 渲染单个表单项
  renderFormItem = ({ item, layout, getFieldDecorator }) => {
    const {
      col = 24, label, key, required, component, options = {}, rules
    } = item;
    return (
      <Col span={col} key={key}>
        <FormItem label={label} {...layout}>
          {getFieldDecorator(key, {
            ...options,
            rules: rules || [{ required, message: `${label}为空` }],
          })(component || <Input />)}
        </FormItem>
      </Col>
    );
  }

  render() {
    const {
      items, layout = defaultFormItemLayout, form: { getFieldDecorator }
    } = this.props;
    return (
      <Form style={{ overflow: 'hidden' }}>
        {items.map(
          item => this.renderFormItem({ item, layout, getFieldDecorator })
        )}
      </Form>
    );
  }
}

export default Form.create()(BaseForm);
