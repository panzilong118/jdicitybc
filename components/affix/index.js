/**
 * @author: chenyanhua
 * @version: 3.4.1
 */
import * as React from 'react';

import * as ReactDOM from 'react-dom';

import PropTypes from 'prop-types';

import addEventListener from 'rc-util/lib/Dom/addEventListener';

import classNames from 'classnames';

import shallowequal from 'shallowequal';

import omit from 'omit.js';

import getScroll from '../_util/getScroll';

import { throttleByAnimationFrameDecorator } from '../_util/throttleByAnimationFrame';

function getTargetRect(target) {
    return target !== window ? target.getBoundingClientRect() : { top: 0, left: 0, bottom: 0 };
}

function getOffset(element, target) {
    const elemRect = element.getBoundingClientRect();
    const targetRect = getTargetRect(target);

    const scrollTop = getScroll(target, true);
    const scrollLeft = getScroll(target, false);

    const docElem = window.document.body;
    const clientTop = docElem.clientTop || 0;
    const clientLeft = docElem.clientLeft || 0;

    return {
        top: elemRect.top - targetRect.top + scrollTop - clientTop,
        left: elemRect.left - targetRect.left + scrollLeft - clientLeft,
        width: elemRect.width,
        height: elemRect.height
    };
}

function noop() {}

function getDefaultTarget() {
    return typeof window !== 'undefined' ? window : null;
}

// Affix
export const AffixProps = {
    /**
     * 距离窗口顶部达到指定偏移量后触发
     */
    offsetTop: PropTypes.number,
    offset: PropTypes.number,
    /** 距离窗口底部达到指定偏移量后触发 */
    offsetBottom: PropTypes.number,
    style: PropTypes.object,
    /** 固定状态改变时触发的回调函数 */
    onChange: PropTypes.func,
    /** 设置 Affix 需要监听其滚动事件的元素，值为一个返回对应 DOM 元素的函数 */
    target: PropTypes.func,
    prefixCls: PropTypes.string
};

export default class Affix extends React.Component {
    static propTypes = {
        ...AffixProps,
        offsetTop: PropTypes.number,
        offsetBottom: PropTypes.number,
        target: PropTypes.func
    };
    
    constructor(props) {
        super(props);

        this.scrollEvent = undefined;
        this.resizeEvent = undefined;
        this.timeout = undefined;

        this.events = [
            'resize',
            'scroll',
            'touchstart',
            'touchmove',
            'touchend',
            'pageshow',
            'load'
        ];

        this.eventHandlers = {};

        this.state = {
            affixStyle: undefined,
            placeholderStyle: undefined
        };

        this.fixedNode = undefined;
        this.placeholderNode = undefined;
    }

    setAffixStyle(e, affixStyle) {
        const { onChange = noop, target = getDefaultTarget } = this.props;
        const originalAffixStyle = this.state.affixStyle;
        const isWindow = target() === window;
        if (e.type === 'scroll' && originalAffixStyle && affixStyle && isWindow) {
            return;
        }
        if (shallowequal(affixStyle, originalAffixStyle)) {
            return;
        }
        this.setState({ affixStyle }, () => {
            const affixed = !!this.state.affixStyle;
            if ((affixStyle && !originalAffixStyle) || (!affixStyle && originalAffixStyle)) {
                onChange(affixed);
            }
        });
    }

    setPlaceholderStyle(placeholderStyle) {
        const originalPlaceholderStyle = this.state.placeholderStyle;
        if (shallowequal(placeholderStyle, originalPlaceholderStyle)) {
            return;
        }
        this.setState({ placeholderStyle });
    }

    syncPlaceholderStyle(e) {
        const { affixStyle } = this.state;
        if (!affixStyle) {
            return;
        }
        this.placeholderNode.style.cssText = '';
        this.setAffixStyle(e, {
            ...affixStyle,
            width: this.placeholderNode.offsetWidth,
        });
        this.setPlaceholderStyle({
            width: this.placeholderNode.offsetWidth,
        });
    }

    @throttleByAnimationFrameDecorator()
    updatePosition(e) {
        const { offsetBottom, offset, target = getDefaultTarget } = this.props;
        let { offsetTop } = this.props;
        const targetNode = target();

        // Backwards support
        offsetTop = offsetTop || offset;
        const scrollTop = getScroll(targetNode, true);
        const affixNode = ReactDOM.findDOMNode(this);
        const elemOffset = getOffset(affixNode, targetNode);
        const elemSize = {
            width: this.fixedNode.offsetWidth,
            height: this.fixedNode.offsetHeight
        };

        const offsetMode = {
            top: false,
            bottom: false
        };
        // Default to `offsetTop=0`.
        if (typeof offsetTop !== 'number' && typeof offsetBottom !== 'number') {
            offsetMode.top = true;
            offsetTop = 0;
        } else {
            offsetMode.top = typeof offsetTop === 'number';
            offsetMode.bottom = typeof offsetBottom === 'number';
        }

        const targetRect = getTargetRect(targetNode);
        const targetInnerHeight = targetNode.innerHeight || targetNode.clientHeight;
        if (scrollTop > elemOffset.top - offsetTop && offsetMode.top) {
            // Fixed Top
            const width = elemOffset.width;
            const top = targetRect.top + offsetTop;
            this.setAffixStyle(e, {
                position: 'fixed',
                top,
                left: targetRect.left + elemOffset.left,
                width,
            });
            this.setPlaceholderStyle({
                width,
                height: elemSize.height,
            });
        } else if (scrollTop < elemOffset.top + elemSize.height + offsetBottom - targetInnerHeight && offsetMode.bottom) {
            // Fixed Bottom
            const targetBottomOffet = targetNode === window ? 0 : (window.innerHeight - targetRect.bottom);
            const width = elemOffset.width;
            this.setAffixStyle(e, {
                position: 'fixed',
                bottom: targetBottomOffet + offsetBottom,
                left: targetRect.left + elemOffset.left,
                width,
            });
            this.setPlaceholderStyle({
                width,
                height: elemOffset.height,
            });
        } else {
            const { affixStyle } = this.state;
            if (e.type === 'resize' && affixStyle && affixStyle.position === 'fixed' && affixNode.offsetWidth) {
                this.setAffixStyle(e, { ...affixStyle, width: affixNode.offsetWidth });
            } else {
                this.setAffixStyle(e, null);
            }
            this.setPlaceholderStyle(null);
        }

        if (e.type === 'resize') {
            this.syncPlaceholderStyle(e);
        }
    }

    componentDidMount() {
        const target = this.props.target || getDefaultTarget;
        // Wait for parent component ref has its value
        this.timeout = setTimeout(() => {
            this.setTargetEventListeners(target);
        });
    }

    componentWillReceiveProps(nextProps) {
        if (this.props.target !== nextProps.target) {
            this.clearEventListeners();
            this.setTargetEventListeners(nextProps.target);

            // Mock Event object.
            this.updatePosition({});
        }
    }

    componentWillUnmount() {
        this.clearEventListeners();
        clearTimeout(this.timeout);
        this.updatePosition.cancel();
    }

    setTargetEventListeners(getTarget) {
        const target = getTarget();
        if (!target) {
            return;
        }
        this.clearEventListeners();

        this.events.forEach((eventName) => {
            this.eventHandlers[eventName] = addEventListener(target, eventName, this.updatePosition);
        });
    }

    clearEventListeners() {
        this.events.forEach((eventName) => {
            const handler = this.eventHandlers[eventName];
            if (handler && handler.remove) {
                handler.remove();
            }
        });
    }

    saveFixedNode = (node) => {
        this.fixedNode = node;
    }

    savePlaceholderNode = (node) => {
        this.placeholderNode = node;
    }

    render() {
        const className = classNames({
            [this.props.prefixCls || 'jc-affix']: this.state.affixStyle
        });

        const props = omit(this.props, ['prefixCls', 'offsetTop', 'offsetBottom', 'target', 'onChange']);
        const placeholderStyle = { ...this.state.placeholderStyle, ...this.props.style };
        return (
            <div {...props} style={placeholderStyle} ref={this.savePlaceholderNode}>
                <div className={className} ref={this.saveFixedNode} style={this.state.affixStyle}>
                    {this.props.children}
                </div>
            </div>
        );
    }
}
