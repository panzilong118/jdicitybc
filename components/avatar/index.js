/**
 * Created by gaoqingli on 2018/4/13.
 */
import * as React from 'react';

import * as ReactDOM from 'react-dom';

import classNames from 'classnames';

import Icon from '../icon';

export default class Avatar extends React.Component {
  static defaultProps = {
      prefixCls: 'jc-avatar',
      shape: 'circle',
      size: 'default',
  };

  // static avatarChildren;

  constructor(props) {
      super(props);
      this.state = {
          scale: 1,
          isImgExist: true,
      };
      this.avatarChildren = undefined;
  }

  componentDidMount() {
      this.setScale();
  }

  componentDidUpdate(prevProps, prevState) {
      if (prevProps.children !== this.props.children
        || (prevState.scale !== this.state.scale && this.state.scale === 1)) {
          this.setScale();
      }
  }

  setScale = () => {
      const childrenNode = this.avatarChildren;
      if (childrenNode) {
          const childrenWidth = childrenNode.offsetWidth;
          const avatarWidth = ReactDOM.findDOMNode(this).getBoundingClientRect().width;
          // add 4px gap for each side to get better performance
          if (avatarWidth - 8 < childrenWidth) {
              this.setState({
                  scale: (avatarWidth - 8) / childrenWidth,
              });
          } else {
              this.setState({
                  scale: 1,
              });
          }
      }
  }

  handleImgLoadError = () => this.setState({ isImgExist: false });

  render() {
      const {
          prefixCls, shape, size, src, icon, className, ...others
      } = this.props;

      const sizeCls = classNames({
          [`${prefixCls}-lg`]: size === 'large',
          [`${prefixCls}-sm`]: size === 'small',
      });

      const classString = classNames(prefixCls, className, sizeCls, {
          [`${prefixCls}-${shape}`]: shape,
          [`${prefixCls}-image`]: src && this.state.isImgExist,
          [`${prefixCls}-icon`]: icon,
      });

      let { children } = this.props;
      if (src && this.state.isImgExist) {
          children = (
              <img
                  alt=''
                  src={src}
                  onError={this.handleImgLoadError}
              />
          );
      } else if (icon) {
          children = <Icon type={icon} />;
      } else {
          const childrenNode = this.avatarChildren;
          if (childrenNode || this.state.scale !== 1) {
              const childrenStyle: React.CSSProperties = {
                  msTransform: `scale(${this.state.scale})`,
                  WebkitTransform: `scale(${this.state.scale})`,
                  transform: `scale(${this.state.scale})`,
                  position: 'absolute',
                  display: 'inline-block',
                  left: `calc(50% - ${Math.round(childrenNode.offsetWidth / 2)}px)`,
              };
              children = (
                  <span
                      className={`${prefixCls}-string`}
                      ref={(span) => { this.avatarChildren = span; }}
                      style={childrenStyle}
                  >
                      {children}
                  </span>
              );
          } else {
              children = (
                  <span
                      className={`${prefixCls}-string`}
                      ref={(span) => { this.avatarChildren = span; }}
                  >
                      {children}
                  </span>
              );
          }
      }
      return (
          <span {...others} className={classString}>
              {children}
          </span>
      );
  }
}
