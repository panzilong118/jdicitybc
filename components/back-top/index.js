import * as React from 'react';

import PropTypes from 'prop-types';

import Animate from 'rc-animate';

import addEventListener from 'rc-util/lib/Dom/addEventListener';

import classNames from 'classnames';

import omit from 'omit.js';

import getScroll from '../_util/getScroll';

import getRequestAnimationFrame from '../_util/getRequestAnimationFrame';

const reqAnimFrame = getRequestAnimationFrame();

const easeInOutCubic = (t, b, c, d) => {
    const cc = c - b;
    t /= d / 2;
    if (t < 1) {
        return cc / 2 * t * t * t + b;
    } else {
        return cc / 2 * ((t -= 2) * t * t + 2) + b;
    }
};

function noop() { }

function getDefaultTarget() {
    return window;
}

export const BackTopProps = {
    visibilityHeight: PropTypes.number, // 滚动高度达到此参数值才出现 BackTop
    onClick: PropTypes.func, // 点击按钮的回调函数
    target: PropTypes.any, // 设置需要监听其滚动事件的元素，值为一个返回对应 DOM 元素的函数,即说明：可以设置div等元素作为滚动框
    prefixCls: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    right: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    bottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
}

export default class BackTop extends React.Component {
    static defaultProps = {
        visibilityHeight: 400,
        style: {},
        className: ''
    };
    static propTypes = { ...BackTopProps };

    constructor(props) {
        super(props);
        this.state = {
            visible: false
        };
        this.scrollEvent = undefined;
    }

    /**
     * 获取当前的ScrollTop值
     */
    getCurrentScrollTop = () => {
        const getTarget = this.props.target || getDefaultTarget;
        const targetNode = getTarget();
        if (targetNode === window) {
            return window.pageYOffset || document.body.scrollTop || document.documentElement.scrollTop;
        }
        return targetNode.scrollTop;
    }

    /**
     * 滚动到顶部，有动画效果
     */
    scrollToTop = (e) => {
        const scrollTop = this.getCurrentScrollTop();
        const startTime = Date.now();
        const frameFunc = () => {
            const timestamp = Date.now();
            const time = timestamp - startTime;
            this.setScrollTop(easeInOutCubic(time, scrollTop, 0, 450));
            if (time < 450) {
                reqAnimFrame(frameFunc);
            }
        };
        reqAnimFrame(frameFunc);
        (this.props.onClick || noop)(e);
    }

    /**
     * 滚动到指定位置
     * @param {*滚动距离} value 
     */
    setScrollTop(value) {
        const getTarget = this.props.target || getDefaultTarget;
        const targetNode = getTarget();
        if (targetNode === window) {
            document.body.scrollTop = value;
            document.documentElement.scrollTop = value;
        } else {
            targetNode.scrollTop = value;
        }
    }

    /**
     * 处理Backtop的显示或隐藏
     */
    handleScroll = () => {
        const { visibilityHeight, target = getDefaultTarget } = this.props;
        const scrollTop = getScroll(target(), true);
        this.setState({
            visible: scrollTop > visibilityHeight
        });
    }

    componentDidMount() {
        // 获取BackTop处理的目标元素，可是Widow，也可以是divD等元素
        const getTarget = this.props.target || getDefaultTarget;
        // 绑定滚动监听事件
        this.scrollEvent = addEventListener(getTarget(), 'scroll', this.handleScroll);
        // 处理滚动事件
        this.handleScroll();
    }

    /**
     * 移除滚动监听事件
     */
    componentWillUnmount() {
        if (this.scrollEvent) {
            this.scrollEvent.remove();
        }
    }

    render() {
        const { prefixCls = 'jc-back-top', className = '', children, right, bottom } = this.props;
        let { style = {} } = this.props;
        if (right && !style.right) {
            style.right = right;
        }
         if (bottom && !style.bottom) {
            style.bottom = bottom;
        }
        const classString = classNames(prefixCls, className);

        const defaultElement = (
            <div className={`${prefixCls}-content`}>
                <div className={`${prefixCls}-icon`} />
            </div>
        );

        // fix https://fb.me/react-unknown-prop
        const divProps = omit(this.props, [
            'style',
            'right',
            'bottom',
            'prefixCls',
            'className',
            'children',
            'visibilityHeight',
            'target',
        ]);

        const backTopBtn = this.state.visible ? (
            <div {...divProps} style={style} className={classString} onClick={this.scrollToTop}>
                {children || defaultElement}
            </div>
        ) : null;

        return (
            <Animate component="" transitionName="fade">
                {backTopBtn}
            </Animate>
        );
    }
}
