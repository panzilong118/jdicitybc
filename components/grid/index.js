/**
 * Created by huangxiao3 on 2018/3/16.
 */
import Row from './row';
import Col from './col';
export { Row, Col, };