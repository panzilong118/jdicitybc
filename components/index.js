export { default as Cinput } from './Cinput';

export { default as Flow } from './Flow';

export { default as Tooltip } from './Tooltip';

export { default as CodeMirror } from './CodeMirror';

export { default as Form } from './BaseForm';

export { default as Tab } from './Tab';
