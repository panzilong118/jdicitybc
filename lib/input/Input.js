'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _omit = require('omit.js');

var _omit2 = _interopRequireDefault(_omit);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// eslint校验提示变量没有被使用，暂时注释掉
// import Group from './Group';
// import Search from './Search';
// import TextArea from './TextArea';

function fixControlledValue(value) {
    if (typeof value === 'undefined' || value === null) {
        return '';
    }
    return value;
}

var Input = function (_React$Component) {
    _inherits(Input, _React$Component);

    function Input(props) {
        _classCallCheck(this, Input);

        var _this = _possibleConstructorReturn(this, (Input.__proto__ || Object.getPrototypeOf(Input)).call(this, props));

        _this.handleKeyDown = function (e) {
            var _this$props = _this.props,
                onPressEnter = _this$props.onPressEnter,
                onKeyDown = _this$props.onKeyDown;

            if (e.keyCode === 13 && onPressEnter) {
                onPressEnter(e);
            }
            if (onKeyDown) {
                onKeyDown(e);
            }
        };

        _this.saveInput = function (node) {
            _this.input = node;
        };

        _this.input = {};
        return _this;
    }

    _createClass(Input, [{
        key: 'focus',
        value: function focus() {
            this.input.focus();
        }
    }, {
        key: 'blur',
        value: function blur() {
            this.input.blur();
        }
    }, {
        key: 'getInputClassName',
        value: function getInputClassName() {
            var _classNames;

            var _props = this.props,
                prefixCls = _props.prefixCls,
                size = _props.size,
                disabled = _props.disabled;

            return (0, _classnames2.default)(prefixCls, (_classNames = {}, _defineProperty(_classNames, prefixCls + '-sm', size === 'small'), _defineProperty(_classNames, prefixCls + '-lg', size === 'large'), _defineProperty(_classNames, prefixCls + '-disabled', disabled), _classNames));
        }
    }, {
        key: 'renderLabeledInput',
        value: function renderLabeledInput(children) {
            var _classNames3;

            var props = this.props;
            // Not wrap when there is not addons
            if (!props.addonBefore && !props.addonAfter) {
                return children;
            }

            var wrapperClassName = props.prefixCls + '-group';
            var addonClassName = wrapperClassName + '-addon';
            var addonBefore = props.addonBefore ? React.createElement(
                'span',
                { className: addonClassName },
                ' ',
                props.addonBefore,
                ' '
            ) : null;

            var addonAfter = props.addonAfter ? React.createElement(
                'span',
                { className: addonClassName },
                ' ',
                props.addonAfter,
                ' '
            ) : null;

            var className = (0, _classnames2.default)(props.prefixCls + '-wrapper', _defineProperty({}, wrapperClassName, addonBefore || addonAfter));

            var groupClassName = (0, _classnames2.default)(props.prefixCls + '-group-wrapper', (_classNames3 = {}, _defineProperty(_classNames3, props.prefixCls + '-group-wrapper-sm', props.size === 'small'), _defineProperty(_classNames3, props.prefixCls + '-group-wrapper-lg', props.size === 'large'), _classNames3));

            // Need another wrapper for changing display:table to display:inline-block
            // and put style prop in wrapper
            if (addonBefore || addonAfter) {
                return React.createElement(
                    'span',
                    {
                        className: groupClassName,
                        style: props.style
                    },
                    React.createElement(
                        'span',
                        { className: className },
                        addonBefore,
                        React.cloneElement(children, { style: null }),
                        addonAfter
                    )
                );
            }
            return React.createElement(
                'span',
                { className: className },
                addonBefore,
                children,
                addonAfter
            );
        }
    }, {
        key: 'renderLabeledIcon',
        value: function renderLabeledIcon(children) {
            var _classNames4;

            var props = this.props;

            if (!('prefix' in props || 'suffix' in props)) {
                return children;
            }

            var prefix = props.prefix ? React.createElement(
                'span',
                { className: props.prefixCls + '-prefix' },
                ' ',
                props.prefix,
                ' '
            ) : null;

            var suffix = props.suffix ? React.createElement(
                'span',
                { className: props.prefixCls + '-suffix' },
                ' ',
                props.suffix,
                ' '
            ) : null;

            var affixWrapperCls = (0, _classnames2.default)(props.className, props.prefixCls + '-affix-wrapper', (_classNames4 = {}, _defineProperty(_classNames4, props.prefixCls + '-affix-wrapper-sm', props.size === 'small'), _defineProperty(_classNames4, props.prefixCls + '-affix-wrapper-lg', props.size === 'large'), _classNames4));
            return React.createElement(
                'span',
                {
                    className: affixWrapperCls,
                    style: props.style
                },
                prefix,
                React.cloneElement(children, { style: null, className: this.getInputClassName() }),
                suffix
            );
        }
    }, {
        key: 'renderInput',
        value: function renderInput() {
            var _props2 = this.props,
                value = _props2.value,
                className = _props2.className;
            // Fix https://fb.me/react-unknown-prop

            var otherProps = (0, _omit2.default)(this.props, ['prefixCls', 'onPressEnter', 'addonBefore', 'addonAfter', 'prefix', 'suffix']);

            if ('value' in this.props) {
                otherProps.value = fixControlledValue(value);
                // Input elements must be either controlled or uncontrolled,
                // specify either the value prop, or the defaultValue prop, but not both.
                delete otherProps.defaultValue;
            }
            return this.renderLabeledIcon(React.createElement('input', _extends({}, otherProps, { className: (0, _classnames2.default)(this.getInputClassName(), className), onKeyDown: this.handleKeyDown, ref: this.saveInput })));
        }
    }, {
        key: 'render',
        value: function render() {
            return this.renderLabeledInput(this.renderInput());
        }
    }]);

    return Input;
}(React.Component);

Input.defaultProps = {
    prefixCls: 'jc-input',
    type: 'text',
    disabled: false
};
Input.propTypes = {
    type: _propTypes2.default.string,
    id: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
    size: _propTypes2.default.oneOf(['small', 'default', 'large']),
    maxLength: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
    disabled: _propTypes2.default.bool,
    value: _propTypes2.default.any,
    defaultValue: _propTypes2.default.any,
    className: _propTypes2.default.string,
    addonBefore: _propTypes2.default.node,
    addonAfter: _propTypes2.default.node,
    prefixCls: _propTypes2.default.string,
    autosize: _propTypes2.default.oneOfType([_propTypes2.default.bool, _propTypes2.default.object]),
    onPressEnter: _propTypes2.default.func,
    onKeyDown: _propTypes2.default.func,
    onKeyUp: _propTypes2.default.func,
    onFocus: _propTypes2.default.func,
    onBlur: _propTypes2.default.func,
    prefix: _propTypes2.default.node,
    suffix: _propTypes2.default.node
};
exports.default = Input;
//# sourceMappingURL=Input.js.map