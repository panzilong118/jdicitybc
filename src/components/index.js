export OperatingOnlyTop from './OperatingOnlyTop/OperatingOnlyTop';
export Operating from './Operating/Operating';
export Shop from './Shop/Shop';
export ShopToolBar from './ShopToolBar/ShopToolBar';
export Buyer from './Buyer/Buyer';
export Message from './Message/Message';
export Customer from './Customer/Customer';
export CustomerMini from './CustomerMini/CustomerMini';
export UploadImg from './UploadImg/UploadImg';
export FuncPermission from './FuncPermission/FuncPermission';
export MoneyConvert from './MoneyConvert/MoneyConvert';
export LinkPermission from './FuncPermission/LinkPermission';
export SubAccount from './SubAccount/SubAccount';
export MenuLinkPermission from './FuncPermission/MenuLinkPermission';
export RiskUser from './RiskUser/RiskUser';
export UploadSelect from './UploadSelect/UploadSelect';
export Loading from './Loading/Loading';
export TableItemInfo from './TableItemInfo/TableItemInfo';
export JcBreadCrumb from './JcBreadCrumb/JcBreadCrumb';
export Price from './Price/Price';
export ItemInfo from './ItemInfo/ItemInfo';
export JcRangePicker from './JcRangePicker/JcRangePicker';
export StatusText from './StatusText/StatusText';
export {GridView,GridViewLayout} from './GridView/GridView';
export CustomerRenderHtml from './CustomerRenderHtml/CustomerRenderHtml';
export Crumbs from './Crumbs/CrumbsWrap';

export App from './AppConfig/App';
export AppConfigHoc from './AppConfig/AppConfigHoc';

export SideCustomer from './SideCustomer/SideCustomer';
export CreatePage from './CreatePage/CreatePageWrap';
