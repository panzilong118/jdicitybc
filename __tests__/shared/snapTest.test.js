/**
 * Created by huangxiao3 on 2018/3/13.
 */
import * as React from "react";
import renderer from 'react-test-renderer';
import Input from '../../components/input';
import { Col, Row } from '../../components/grid';
describe('<Input/>', () => {
    it('demo test Snapshot', () => {
        // const componentOri = renderer.create(<Input/>);
        // let snapshotOri = componentOri.toJSON();
        // expect(snapshotOri).toMatchSnapshot();

        const componentAddon = renderer.create(<Input addonBefore="Http://" addonAfter=".com" defaultValue="mysite" />);
        let snapshotAddon = componentAddon.toJSON();
        expect(snapshotAddon).toMatchSnapshot();

        const componentGrid = renderer.create(
            <Row gutter={16}>
                <Col span={8}>
                    123
                </Col>
                <Col span={8}>
                    456
                </Col>
                <Col span={8}>
                    789
                </Col>
            </Row>
        );
        let snapsGrid = componentGrid.toJSON();
        expect(snapsGrid).toMatchSnapshot();


        // snapshot.props.focus();
        // snapshot = component.toJSON();
        // expect(snapshot).toMatchSnapshot();

    });
});