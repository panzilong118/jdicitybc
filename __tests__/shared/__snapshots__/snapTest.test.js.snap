// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> demo test Snapshot 1`] = `
<span
  className="jc-input-group-wrapper"
  style={undefined}
>
  <span
    className="jc-input-wrapper jc-input-group"
  >
    <span
      className="jc-input-group-addon"
    >
       
      Http://
       
    </span>
    <input
      className="jc-input"
      defaultValue="mysite"
      disabled={false}
      onKeyDown={[Function]}
      style={null}
      type="text"
    />
    <span
      className="jc-input-group-addon"
    >
       
      .com
       
    </span>
  </span>
</span>
`;

exports[`<Input/> demo test Snapshot 2`] = `
<div
  className="jc-row"
  style={
    Object {
      "marginLeft": -8,
      "marginRight": -8,
    }
  }
>
  <div
    className="jc-col-8"
    style={
      Object {
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    123
  </div>
  <div
    className="jc-col-8"
    style={
      Object {
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    456
  </div>
  <div
    className="jc-col-8"
    style={
      Object {
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    789
  </div>
</div>
`;
